bundles/org.openhab.core.crypto.cryptostore/src/main/java/org/openhab/core/crypto/cryptostore/CryptoStore.java

```diff
@@ -0,0 +1,328 @@
+        Certificate caCert = CertificateFactory.getInstance("X.509")
+                .generateCertificate(new ByteArrayInputStream(Base64.getDecoder().decode(this.caCert.getBytes())));
+        return caCert;
+    }
+
+    public void setAlias(String alias) {
+        this.alias = alias;
+    }
+
+    public String getAlias() {
+        return this.alias;
+    }
+
+    public void setCAAlias(String caAlias) {
+        this.caAlias = caAlias;
+    }
+
+    public String getCAAlias() {
+        return this.caAlias;
+    }
+
+    public void setAlgorithm(String keystoreAlgorithm) {
+        this.keystoreAlgorithm = keystoreAlgorithm;
+    }
+
+    public String getAlgorithm() {
+        return this.keystoreAlgorithm;
+    }
+
+    public void setKeySize(int keySize) {
+        this.keySize = keySize;
+    }
+
+    public int getKeySize() {
+        return this.keySize;
+    }
+
+    public void setKeyLength(int keyLength) {
+        this.keyLength = keyLength;
+    }
+
+    public int getKeyLength() {
+        return this.keyLength;
+    }
+
+    public void setDistName(String distName) {
+        this.distName = distName;
+    }
+
+    public String getDistName() {
+        return this.distName;
+    }
+
+    public void setKeystoreFileName(String keystoreFileName) {
+        this.keystoreFileName = keystoreFileName;
+    }
+
+    public String getKeystoreFileName() {
+        return this.keystoreFileName;
+    }
+
+    public void setKeys(String privKey, byte[] keyString, String cert) throws GeneralSecurityException, Exception {
+        setPrivKey(privKey, keyString);
+        setCert(cert);
+    }
+
+    public void setKeyStore(String keystoreFileName) {
+        this.keystoreFileName = keystoreFileName;
+    }
+
+    public void loadFromKeyStore(String keystoreFileName, String keystorePassword, byte[] keyString)
+            throws GeneralSecurityException, IOException, Exception {
+        this.keystoreFileName = keystoreFileName;
+        loadFromKeyStore(keystorePassword, keyString);
+    }
+
+    public void loadFromKeyStore(String keystorePassword, byte[] keyString)
+            throws GeneralSecurityException, IOException, Exception {
+        Key key = convertByteToKey(keyString);
+        KeyStore keystore = KeyStore.getInstance("JKS");
+        FileInputStream keystoreInputStream = new FileInputStream(this.keystoreFileName);
+        keystore.load(keystoreInputStream, keystorePassword.toCharArray());
+        byte[] byteKey = keystore.getKey(this.alias, keystorePassword.toCharArray()).getEncoded();
+        this.privKey = encrypt(new String(Base64.getEncoder().encode(byteKey)), key);
+        setCert(keystore.getCertificate(this.alias));
+        String caAlias = this.caAlias;
+        Certificate caCert = keystore.getCertificate(caAlias);
+        if (caCert != null) {
+            setCaCert(caCert);
+        }
+    }
+
+    public KeyStore getKeyStore(String keystorePassword, byte[] keyString)
```

Comment:
Is this for creating a new keystore? Would be better to name it accordingly

Reply:
Yes, this generates a new keystore.  What would you suggest to adjust the name too?  getKeyStore seemed clean.

Reply:
`getSomething` is usually used to get a field, `createKeyStore` would better reflect the purpose.

Reply:
That's fair.  In my original thinking, since we were never actually storing a KeyStore, a get seemed like a good facade to the binding developer where we were returning it.  I can change this easily enough.